src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -679,12 +671,14 @@ def _runtool(self, main, tool_name, args, distribution, tgt=None, input_files=tu
 
         epr = ExecuteProcessRequest(
           argv=tuple(cmd),
-          env=dict(),
           input_files=input_files_directory_digest,
           output_files=tuple(),
           output_directories=(output_dir,),
           timeout_seconds=15*60,
-          description='run {} for {}'.format(tool_name, tgt)
+          description='run {} for {}'.format(tool_name, tgt),
+          # TODO: These should always be unicodes
```

Comment:
Hm so depends on the interpreter we're using and the values of those two variables. 

It looks the intended types are `tool_name: str, tgt: Optional[str]`? (The backported `str`, i.e. unicode).

When using Python 2, we can mix byte strings and unicode freely. So, if either `tool_name` or `tgt` are bytes, nothing will complain, which it sounds like we do not want to allow this mixing? In Python 3, we'll get an exception for trying to use `.format()` on bytes and for mixing unicode and bytes.

--

`six` won't actually be very helpful here. We have two options to ensure it's unicode:
```python
tool_name = ensure_text(tool_name)  # will coerce into unicode
```
```python
from builtins import str

assert isinstance(tool_name, str)
```

Also a little weirdness that `tgt` is `Optional`. It looks like it will print the string literal `"None"` if the value is `None`. Is that intended?